### ISSUE TITLE
LogicalExpressions Split Between Multiple Lines Don't Indent Consistently After Fix

### ISSUE THREAD

**Issue by epmatsw** [eslint/eslint#8443]
**Tell us about your environment**

* **ESLint Version:** 4.0.0-alpha.0
* **Node Version:** 7.7.4
* **npm Version:** 4.5.0

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

```
{
    "extends": "eslint:recommended",
    "rules": {
        "indent": 2,
    }
}
```

**What did you do? Please include the actual source code causing the issue.**

```js
(function () {
    if (
       a === b &&
       b === c
    ) {
        return d;
    }
})();
```
Importantly, note the 7 leading spaces before both lines in the LogicalExpression.

**What did you expect to happen?**
Both lines would be indented to 8 spaces using --fix.
**What actually happened? Please include the actual, raw output from ESLint.**
Only the first line gets adjusted.

```
(function () {
    if (
        a === b &&
       b === c
    ) {
        return d;
    }
})();
```

**Comment by epmatsw** [comment]
This seems to apply to multi-line things in general:

Input:
```
const a = {
    b: doAThing(
     'a' +
         'd' +
     'e'
    ),
};
```

Output:
```
const a = {
    b: doAThing(
        'a' +
         'd' +
     'e'
    ),
};

```

**Comment by not-an-aardvark** [comment]
Thanks for the report. At the moment, the rule always ignores multiline binary expressions such as `a + b` (the old version of the rule also did this), so it isn't able to determine that `b === c` is indented incorrectly in your example.

If I recall correctly, I decided to ignore this type of code for backwards compatibility when I thought the changes to `indent` would be released in a minor version. It might be possible to check the indentation of this type of code now.

**Comment by epmatsw** [comment]
Yeah, I guess it would be fine to ignore it, I just wish that if it were part of a multiline expression that that first line would be ignored if the whole thing isn't going to get checked for indentation. As it is, the fixer basically goes through and takes "okay but not perfect" code and turns it into the "kind of more correct but now inconsistent" examples above, which is almost worse...

**Comment by not-an-aardvark** [comment]
After looking into it a bit, it seems like it will be difficult to do this in a way that everyone would agree with -- there seem to be a number of different styles:

```js
// everything aligned
if (
  foo &&
  bar &&
  baz
) {
  qux
}
```

```js
// offset by 1
return foo &&
    bar &&
    baz;
```

This isn't a blocker for 4.0.0, so I'm going to remove it from the milestone. We can always add an opt-in to checking these expressions in a semver-minor release later on.

**Comment by epmatsw** [comment]
I guess in that case I'd really prefer that this case wasn't included in `--fix`. If the initial line is flagged but skipped by the fix, a developer will see the error and can go back and fix all of it manually. If it gets fixed, we'll never catch the now weirdly-indented subsequent lines...

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get enough support from the team and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after a long time tend to never do it, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
